File: 8kyu/(8 kyu) Reversed Words/(8 kyu) Reversed Words.py
# #1
# def reverseWords(str):
#     arr = str.split()
#     rev = arr[::-1]
#     return " ".join(rev)
# #2
# def reverseWords(str):
#     return " ".join(str.split()[::-1])
# #3


def reverseWords(s): return " ".join(s.split(" ")[::-1])

File: 8kyu/(8 kyu) Reversed Words/(8 kyu) Reversed Words.dart
// 1 Simple solution
String reverseWords(String str) {
  final words = str.split(' ');
  String res = '';
  for (int i = words.length - 1; i >= 0; i--) {
    res += words[i] + ' ';
  }
  return res.trim();
}

// 2 Optimized solution
String reverseWords(String str) {
  final words = str.split(' ');
  final reversedWords = words.reversed;
  return reversedWords.join(' ').trim();
}

// 3 Clever solution
String reverseWords(String str) => str.split(' ').reversed.join(' ').trim();

// 4 Coding golf
reverseWords(s) => s.split(' ').reversed.join(' ');
